### DISCUSSION TITLE
Why doesn't "Remember Me" work through browser restarts?

### DISCUSSION THREAD

**Discussion by lOmicronl** [keycloak/keycloak#22319] [Q&A] [answered]
I'm fairly new to Keycloak, and so configuring a realm is mostly an exercise in feeding Google the right questions.

When I search for how to get a user session to persist even after the browser has been closed, I always find the answer: "that's what the remember me function is for, if you turn it on, users will be able to return to their session even after rebooting their computer."

Except... that doesn't seem to work for me? When I enable the checkbox, and check it on login, then close the browser without logging out, open the browser again, and open the address of the application, the application doesn't remember me at all. Instead I am once more presented with the login mask.

Furthermore, when I login again after re-opening the browser, I don't get the old session back. I get a new one instead. This is despite the fact that the old session is still alive and well. The old session will live the entire six hours worth of "SSO Session Idle Remember Me" that I configured for this test, but it is no longer accessible.

The browser is _not_ set to delete cookies on closing.

Is there something else that I need to configure for Remember Me to work correctly?

**Comment by justin-tay** [comment]
I can confirm that this functionality should work. After configuring `Realm settings > Login > Remember me` there is now a `Remember me` checkbox on the Keycloak login page. If I submit the username and password, Keycloak will log me on. There is now a new `KEYCLOAK_REMEMBER_ME` persistent cookie. The `KEYCLOAK_IDENTITY` and `KEYCLOAK_IDENTITY_LEGACY` cookies are now persistent cookies instead of session cookies. If I close an reopen the browser and navigate to my application it can automatically complete the login flow with Keycloak without Keycloak displaying the login page.

You didn't really describe how you are integrating your application with Keycloak though.

**Comment by lOmicronl** [comment]
Thank you, just having that confirmed helps.

On Monday I will examine these three cookies you named - if they are all there, and if they are persistent. Then I will look at what happens to them when I close the browser.

The application is unfortunately a bit of a blackbox, as it isn't developed inhouse. It forwards authentication to a Keycloak server address we provide in the config file, but I don't know how it works under the hood. However, I can certainly contact the developer if it turns out that all the cookies are working correctly but the session is still not persisting.

**Comment by lOmicronl** [comment, marked as answer]
Upon checking the cookies, I found them all present and persistent. However, the browser (Firefox) was for some reason deleting them upon closing, even though the relevant checkbox in the settings remains unchecked. (Perhaps some company configuration policy is overriding it.)

I was able to add the website as an exception that explicitly allows all cookies. After doing this, the cookies are no longer deleted and now persist across browser restarts.

Thanks again for pointing me in the right direction :)
